### PR TITLE
Prevent infinite loading in validateBankAccount

### PR DESCRIPTION
**Pull request by luacmartins**
cc @chiragsalian 
### Details
Removes `return` statement that kept API calls in an infinite loading state.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/171371

### Tests
1. Create a new Workspace
2. Follow [these steps](https://stackoverflow.com/c/expensify/questions/342/525#525) to add a `PENDING` account.
3. When prompted for the 3 codes, follow [these steps](https://stackoverflow.com/c/expensify/questions/8323).
4. Verify that the button text reappears.

### QA Steps
Internal QA.

### Tested On

- [X] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web

https://user-images.githubusercontent.com/22219519/126534777-41718933-61e7-4faa-9053-6896b1ea6cc5.mov

### CODE REVIEW

File: src/libs/actions/BankAccounts.js [hunk 1]
```diff
@@ -553,7 +553,6 @@ function validateBankAccount(bankAccountID, validateCode) {
             if (response.jsonCode === 200) {
                 Growl.show('Bank Account successfully validated!', CONST.GROWL.SUCCESS, 3000);
                 Navigation.dismissModal();
-                return;
```
Comment (marcaaron) [inline src/libs/actions/BankAccounts.js:556]: Nice catch! 

But I think the intention to early return here is correct as the non 200 case would be error handling logic. 

So we just need to set loading to `false` and `error` to empty string maybe `''`?

```js
    Navigation.dismissModal();
    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: false, error: ''});
    return;
}
```

**Comment by luacmartins** [comment]
Updated!

**Review by marcaaron — APPROVED** [review]

**Comment by MitchExpensify** [comment]
https://user-images.githubusercontent.com/36425901/127045839-9c533e0b-dcbf-45bc-84ea-0a63a35b1e79.mov

It was weird to see "Get Started" as the button in Expensify Card when it should read "Finish Setup". That screen did not auto update after entering the validation amounts. I would have expected to see "Manage Cards" at that point

**Comment by luacmartins** [comment]
Thanks for the input @mitch! I put up [this PR](https://github.com/Expensify/App/pull/4237) to update the button text.

**Comment by isagoico** [comment]
@luacmartins @MitchExpensify Should this issue https://github.com/Expensify/App/pull/4165#issuecomment-886961530 be treated as a deploy blocker or can we check off this PR from the list?

**Comment by luacmartins** [comment]
I think it's fine to check it off the list since it solves the issue at hand and the remaining steps are only copy that needs to be updated. I put up [another PR](https://github.com/Expensify/App/pull/4237) to update the button text.

**Comment by kevinksullivan** [comment]
@luacmartins I ran into the same experience after the `Company Information` step, where now I am stuck on a loading spinner.

image


So I haven't been able to get to the ValidateBankAccount step, but it looks like we may have similar problems at each step of the process... Should I open a separate GH? And would you want to pick it up since you just knocked this one out?

**Comment by kevinksullivan** [comment]
Is there a reason this didn't closed if the PR is deployed to production?

**Comment by luacmartins** [comment]
Not sure, for some reason I can't manually close it either...

**Comment by MitchExpensify** [comment]
Weird, neither can I.

**Pull request merged**